shared_testutil/test_ipld_tree.go

```diff
@@ -68,31 +67,44 @@ func NewTestIPLDTree() TestIPLDTree {
 	}
 
 	var (
-		leafAlpha, leafAlphaLnk         = encode(fnb.CreateString("alpha"))
+		leafAlpha, leafAlphaLnk         = encode(fluent.MustBuild(basicnode.Style.String, func(na fluent.NodeAssembler) { na.AssignString("alpha") }))
 		leafAlphaBlock, _               = blocks.NewBlockWithCid(storage[leafAlphaLnk], leafAlphaLnk.(cidlink.Link).Cid)
-		leafBeta, leafBetaLnk           = encode(fnb.CreateString("beta"))
+		leafBeta, leafBetaLnk           = encode(fluent.MustBuild(basicnode.Style.String, func(na fluent.NodeAssembler) { na.AssignString("beta") }))
 		leafBetaBlock, _                = blocks.NewBlockWithCid(storage[leafBetaLnk], leafBetaLnk.(cidlink.Link).Cid)
-		middleMapNode, middleMapNodeLnk = encode(fnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
-			mb.Insert(knb.CreateString("foo"), vnb.CreateBool(true))
-			mb.Insert(knb.CreateString("bar"), vnb.CreateBool(false))
-			mb.Insert(knb.CreateString("nested"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
-				mb.Insert(knb.CreateString("alink"), vnb.CreateLink(leafAlphaLnk))
-				mb.Insert(knb.CreateString("nonlink"), vnb.CreateString("zoo"))
-			}))
+		middleMapNode, middleMapNodeLnk = encode(fluent.MustBuildMap(basicnode.Style.Map, 3, func(ma fluent.MapAssembler) {
+			nva := ma.AssembleEntry("foo")
```

Comment:
You can actually chain these and cut your lines almost in half. I just made the changes and all the tests still pass:
```suggestion
			ma.AssembleEntry("foo").AssignBool(true)
			ma.AssembleEntry("bar").AssignBool(false)
			ma.AssembleEntry("nested").CreateMap(2, func(ma fluent.MapAssembler) {
				ma.AssembleEntry("alink").AssignLink(leafAlphaLnk)
				ma.AssembleEntry("nonlink").AssignString("zoo")
			})
		}))
		middleMapBlock, _                 = blocks.NewBlockWithCid(storage[middleMapNodeLnk], middleMapNodeLnk.(cidlink.Link).Cid)
		middleListNode, middleListNodeLnk = encode(fluent.MustBuildList(basicnode.Style.List, 4, func(la fluent.ListAssembler) {
			la.AssembleValue().AssignLink(leafAlphaLnk)
			la.AssembleValue().AssignLink(leafAlphaLnk)
			la.AssembleValue().AssignLink(leafBetaLnk)
			la.AssembleValue().AssignLink(leafAlphaLnk)
		}))
		middleListBlock, _    = blocks.NewBlockWithCid(storage[middleListNodeLnk], middleListNodeLnk.(cidlink.Link).Cid)
		rootNode, rootNodeLnk = encode(fluent.MustBuildMap(basicnode.Style.Map, 4, func(ma fluent.MapAssembler) {
			ma.AssembleEntry("plain").AssignString("olde string")
			ma.AssembleEntry("linkedString").AssignLink(leafAlphaLnk)
			ma.AssembleEntry("linkedMap").AssignLink(middleMapNodeLnk)
			ma.AssembleEntry("linkedList").AssignLink(middleListNodeLnk)
		}))
```

Reply:
cool! cc: @warpfork this makes for a wonderfully concise syntax

Reply:
oh thank goodness